ibm/resource_ibm_cis_origin_pool.go

```diff
@@ -2,300 +2,383 @@ package ibm
+	return resourceCISPoolRead(d, meta)
 }
 
-func resourceCISpoolRead(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISPoolRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisGLBPoolClientSession()
 	if err != nil {
 		return err
 	}
-	poolId, cisId, err := convertTftoCisTwoVar(d.Id())
+	poolID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	var pool *v1.Pool
-	pool, err = cisClient.Pools().GetPool(cisId, poolId)
+	cisClient.Crn = core.StringPtr(crn)
+	opt := cisClient.NewGetLoadBalancerPoolOptions(poolID)
+	result, resp, err := cisClient.GetLoadBalancerPool(opt)
 	if err != nil {
-		if checkCisPoolDeleted(d, meta, err, pool) {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[WARN] Error getting zone during PoolRead %v\n", err)
+		log.Printf("[WARN] Create GLB Pools failed %s\n", resp)
 		return err
 	}
 
-	poolObj := *pool
-	d.Set("cis_id", cisId)
-	d.Set("name", poolObj.Name)
-	d.Set("check_regions", poolObj.CheckRegions)
-	d.Set("origins", flattenOrigins(poolObj.Origins))
-	d.Set("description", poolObj.Description)
-	d.Set("enabled", poolObj.Enabled)
-	d.Set("minimum_origins", poolObj.MinOrigins)
-	d.Set("monitor", convertCisToTfTwoVar(poolObj.Monitor, cisId))
-	d.Set("notification_email", poolObj.NotEmail)
-	d.Set("health", poolObj.Health)
-	d.Set("created_on", poolObj.CreatedOn)
-	d.Set("modified_on", poolObj.ModifiedOn)
-
+	poolObj := *result.Result
+	d.Set(cisID, crn)
+	d.Set(cisGLBPoolID, poolObj.ID)
+	d.Set(cisGLBPoolName, poolObj.Name)
+	d.Set(cisGLBPoolOrigins, flattenOrigins(poolObj.Origins))
+	d.Set(cisGLBPoolRegions, poolObj.CheckRegions)
+	d.Set(cisGLBPoolDesc, poolObj.Description)
+	d.Set(cisGLBPoolEnabled, poolObj.Enabled)
+	d.Set(cisGLBPoolNotificationEMail, poolObj.NotificationEmail)
+	d.Set(cisGLBPoolHealthy, poolObj.Healthy)
+	d.Set(cisGLBPoolMinimumOrigins, poolObj.MinimumOrigins)
+	d.Set(cisGLBPoolCreatedOn, poolObj.CreatedOn)
+	d.Set(cisGLBPoolModifiedOn, poolObj.ModifiedOn)
+	if poolObj.Monitor != nil {
+		d.Set(cisGLBPoolMonitor, *poolObj.Monitor)
+	}
 	return nil
 }
 
-func resourceCISpoolUpdate(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
-	if err != nil {
-		return err
-	}
-	poolID, cisID, err := convertTftoCisTwoVar(d.Id())
+func resourceCISPoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisGLBPoolClientSession()
 	if err != nil {
 		return err
 	}
-	poolUpdate := v1.PoolBody{}
-	if d.HasChange("name") || d.HasChange("origins") || d.HasChange("check_regions") || d.HasChange("notification_email") || d.HasChange("monitor") || d.HasChange("enabled") || d.HasChange("minimum_origins") || d.HasChange("description") {
-		if name, ok := d.GetOk("name"); ok {
-			poolUpdate.Name = name.(string)
+	cisID := d.Get(cisID).(string)
```

Comment:
please get the cis and pool id values from d.ID

Reply:
done